WalletWasabi/Blockchain/Analysis/BlockchainAnalyzer.cs

```diff
@@ -107,32 +108,47 @@ private double Intersect(IEnumerable<double> anonsets)
 		return normalizedIntersectionAnonset;
 	}
 
-	private void AnalyzeCoinjoinWalletOutputs(SmartTransaction tx, double startingMixedOutputAnonset, double startingNonMixedOutputAnonset, ISet<HdPubKey> distinctWalletInputPubKeys)
+	private void AnalyzeCoinjoinWalletOutputs(SmartTransaction tx, double startingMixedOutputAnonset, double startingNonMixedOutputAnonset)
 	{
-		var indistinguishableWalletOutputs = tx.WalletOutputs
+		var indistinguishableWalletOutputs = tx.WalletVirtualOutputs
 			.GroupBy(x => x.Amount)
 			.ToDictionary(x => x.Key, y => y.Count());
 
-		var indistinguishableOutputs = tx.Transaction.Outputs
-			.GroupBy(x => x.ScriptPubKey)
-			.ToDictionary(x => x.Key, y => y.Sum(z => z.Value))
-			.GroupBy(x => x.Value)
+		var indistinguishableForeignOutputs = tx.ForeignVirtualOutputs
+			.GroupBy(x => x.Amount)
 			.ToDictionary(x => x.Key, y => y.Count());
 
-		var outputValues = tx.Transaction.Outputs.Select(x => x.Value).OrderByDescending(x => x).ToArray();
-		var secondLargestOutputAmount = outputValues.Distinct().OrderByDescending(x => x).Take(2).Last();
-		bool? isWasabi2Cj = null;
+		var outputValues = tx
+			.WalletVirtualOutputs
+			.Select(x => x.Amount.Satoshi)
+			.Concat(tx.ForeignVirtualOutputs.Select(x => x.Amount.Satoshi))
+			.OrderByDescending(x => x)
+			.ToArray();
+
+		bool isWasabi2Cj =
+					tx.Transaction.Outputs.Count >= 2 // Sanity check.
+					&& tx.Transaction.Inputs.Count >= 50 // 50 was the minimum input count at the beginning of Wasabi 2.
+					&& outputValues.Count(x => StdDenoms.Contains(x)) > tx.Transaction.Outputs.Count * 0.8 // Most of the outputs contains the denomination.
```

Comment:
`tx.Transaction.Outputs.Count` should be `WalletVirtualOutputs` + `ForeignVirtualOutputs` count.